packages/pathfinder/src/pathfinder.ts

```diff
@@ -0,0 +1,400 @@
+    }
+
+    /****************************************
+     * Initialization methods
+     ****************************************/
+
+    /**
+     * Should perform the following actions:
+     *
+     * - loads the list of supported assets
+     */
+    public async init() {
+
+        return Promise.all([
+            this.getSupportedAssets().then(() => this.validateSupportedAssets(this.supportedAssets))
+        ]);
+
+    }
+
+    /**
+     * Loads the list of supported assets from the postgres database
+     */
+    private async getSupportedAssets() {
+
+        let assets = [] as stellarAsset[];
+        const query = {
+            text: 'SELECT id, asset_code, asset_issuer, asset_type from ASSETS'
+        };
+        return pool.query(query)
+            .then((queryResult: any) => queryResult.rows)
+            .then((dbAssets: any) => {
+                assets = dbAssets;
+                this.supportedAssets = dbAssets;
+                return dbAssets;
+            });
+            
+    }
+
+    /**
+     * helper for below
+     * returns a key to be used for lookup in the form of `code-issuer`
+     */
+    private makeKeyPairHelper = (asset_code: string, asset_issuer: string, asset_type: string): string => {
+        const key = (asset_type !== 'native')
+            ? `${asset_code}-${asset_issuer}`
+            : 'XLM';
+        return key;
+    }
+
+
+    /**
+     * Takes in stellarAsset | StellarSdk.Asset 
+     * returns a key to be used for lookup in the form of `code-issuer`
+     */
+    private makeKeyPair = (asset: any): string => {
+        const { asset_code, asset_issuer, asset_type } = asset;
+        return this.makeKeyPairHelper(asset_code, asset_issuer, asset_type);
+    }
+
+    /**
+    * Takes in a StellarSdk.PaymentPathRecord and 
+    * returns a key to be used for lookup in the form of `code-issuer`
+    */
+    private makeKeyPairFromPath = (pathResult: StellarSdk.PaymentPathRecord ): string => {
+        const { source_asset_code, source_asset_issuer, source_asset_type } = pathResult;
+        return this.makeKeyPairHelper(source_asset_code, source_asset_issuer, source_asset_type);
+    }
+
+
+    /**
+     * Ensures that the account belonging to REFERENCE_ACCOUNT_PUBLIC_KEY contains balances
+     * of all of (but also ONLY) the indicated assets
+     *
+     * throws PathfinderInitializationError if incorrectly formatted
+     */
+    private async validateSupportedAssets(assets: [stellarAsset]) {
+        return this.server.loadAccount(Pathfinder.REFERENCE_ACCOUNT_PUBLIC_KEY)
+            .then((account: StellarSdk.AccountRecord) => account.balances)
+            // todo Resolve this type to stellarBalance
+            .then((balances: any) => {
+                if (assets.length !== balances.length) throw PathfinderInitializationError;
+                else {
+                    const asset_dict: { [index: string]: boolean } = {};
+                    assets.map((asset: stellarAsset) => {
```

Comment:
Could we get a few comment words describing what you are doing here. I know what the code is doing, but I want to be sure it matches what you think it is doing.